src/NServiceBus.Core/Saga/TimeoutContext.cs

```diff
@@ -0,0 +1,44 @@
+namespace NServiceBus.Saga
+{
+    using System;
+
+    internal class TimeoutContext : ITimeoutContext
```

Comment:
Discussed this with @johnsimons and @andreasohlund we wanted to go towards a flatter structure without inheritance like we previously had with the send options (which is now changed as well). Let's address this in another PR ok?